backend/app/db/vector_store.py

```
@@ -0,0 +1,41 @@
+            region=AwsRegion.US_EAST_1
+        )
+    )
+else:
+    print(f"Index '{INDEX_NAME}' already exists")
+
+try:
+    # Connect to the index
+    index = pc.Index(INDEX_NAME)
+except Exception as e:
+    raise RuntimeError("Error occured while "
+                       f"connecting to the index {INDEX_NAME}:{e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)